tlsfuzzer/messages.py

```diff
@@ -764,36 +764,43 @@ def new_calculate_mac(mac, seqnumBytes, contentType, data,
 
     return generator
 
-def fuzz_seal(generator, substitutions=None, xors=None):
+def fuzz_encrypted_message(generator, substitutions=None, xors=None):
     """Change arbitrary bytes of the AEAD sealed block"""
     def new_generate(state, self=generator,
                      old_generate=generator.generate,
                      substitutions=substitutions,
                      xors=xors):
-        """Monkey patch to modify sealed AEAD block"""
+        """Monkey patch to modify ciphertext block"""
         msg = old_generate(state)
 
-        old_encryptThenSeal = state.msg_sock._encryptThenSeal
+        old_send = state.msg_sock._recordSocket.send
 
-        self.old_encryptThenSeal = old_encryptThenSeal
+        self.old_send = old_send
 
-        def new_encryptThenSeal(buf, contentType,
-                              old_encryptThenSeal=old_encryptThenSeal,
-                              substitutions=substitutions,
-                              xors=xors):
-            """Monkey patch for the AEAD method of msg socket"""
-            sealed = old_encryptThenSeal(buf, contentType)
+        def new_send(message, padding, old_send=old_send, substitutions=substitutions,
+                     xors=xors):
+            """Monkey patch for the send method of msg socket"""
+
+            """
+                message.data is the encrypted tls record, e.g.
+                message.data = aead_encrypt(plain); defined by aead suite
+                message.data = encrypt(plain + padding) + mac; etm
+                message.data = encrypt(plain + mac + padding); mte
+            """
+            data = message.data
```

Comment:
this is not guaranteed to work, you can get higher level messages, `message.write()` will work for any object that can be passed to socket